### PR TITLE
renderer: Add screenshot support

### PR DESCRIPTION
**Pull request by Macdu**
Add support for taking screenshot on Vita3K.
Screenshot are taken directly from the PS Vita surface, meaning that they use the native resolution (at 8x they will be of size 7860x4352, even if your screen does not support it), it is possible to bind a key to take a screenshot.
This is implemented on both the opengl and vulkan renderer.

### CODE REVIEW

File: vita3k/renderer/src/vulkan/surface_cache.cpp [hunk 1]
```diff
@@ -1218,4 +1218,53 @@ vk::ImageView VKSurfaceCache::sourcing_color_surface_for_presentation(Ptr<const
...
+
+    // this will cause a waitIdle, not an issue
+    vkutil::end_single_time_command(state.device, state.general_queue, state.general_command_pool, cmd_buffer);
+
+    memcpy(frame.data(), temp_buff.mapped_data, frame.size() * 4);
+
+    return frame;
+}
+
 }; // namespace renderer::vulkan
```
Comment (Zangetsu38) [inline vita3k/renderer/src/vulkan/surface_cache.cpp:1270]: if you can fix endline in same time 🙏

Comment (Macdu) [reply vita3k/renderer/src/vulkan/surface_cache.cpp]: done

**Review by EXtremeExploit — APPROVED** [review]
looks nice

File: vita3k/gui/src/controls_dialog.cpp [hunk 2]
```diff
@@ -195,6 +197,7 @@ void draw_controls_dialog(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::TableSetupColumn("button");
         ImGui::TableSetupColumn("mapped_button");
         remapper_button(gui, emuenv, &emuenv.cfg.keyboard_toggle_texture_replacement, lang["toggle_texture_replacement"].c_str());
+        remapper_button(gui, emuenv, &emuenv.cfg.keyboard_take_screenshot, "Take a screenshot");
```
[the remark this answers is not in the document]
Comment (Macdu) [reply vita3k/gui/src/controls_dialog.cpp:200]: Yes, I'll let someone else add some translation and the lang support

**Comment by eisnerguy1** [comment]
Thanks for adding this! 😁 👍🏻 What's the default path where the screenshots are saved? For the life of me I can't seem to find it. I'm on macOS btw.

**Pull request merged**